### PR TITLE
Added ReadRepository and Specification 

### PR DESCRIPTION
**Pull request by TimoHeiten**
(for all query types not just filters)

### CODE REVIEW

File: tests/SpecificationTests.cs [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+            // Given
+            var source = Enumerable.Range(0, 14).Select(x => new TextContainer()).AsQueryable();
+            var spec = new FindNameEqMimiSpecification();
+
+            // When
+            var listResult = await spec.RunAsync(source);
+            var singleResult = await spec.RunSingleAsync(source);
+
+            // Then
+            listResult.Should().HaveCount(7);
```
Comment (Flash0ver) [inline tests/SpecificationTests.cs]: Perhaps we also want to assert that we only have `"Mimi"` in that result set:
`listResult.Should().HaveCount(7).And.OnlyContain(x => x.Name == "Mimi");`

File: tests/SpecificationTests.cs [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+            singleResult.Should().NotBeNull();
```
Comment (Flash0ver) [inline tests/SpecificationTests.cs:26]: Perhaps we additionally want to assert that the `Name` is `"Mimi"` indeed:
`singleResult.Name.Should().Be("Mimi");`

File: src/heitech.efXt/IReadRepository.cs [hunk 3]
```diff
@@ -0,0 +1,16 @@
+using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+
+namespace heitech.efXt
+{
+    ///<summary>
+    /// A Repository that completely relies on the Specification pattern. If you like a more simple one, use the IRepository
+    ///</summary>
+     public interface IReadRepository<TEntity, TId>
```
Comment (Flash0ver) [inline src/heitech.efXt/IReadRepository.cs]: Does it make sense to constrain the type argument `TId` to implement `IEquatable<TId>` and/or `IComparable<TId>`.
I'm not that familiar with the Entity Framework internals. Do keys need to implement (any of) these interfaces?

**Review by Flash0ver — APPROVED** [review]

File: src/heitech.efXt/IReadRepository.cs (hunk 3 above)
Comment (TimoHeiten) [reply src/heitech.efXt/IReadRepository.cs]: No it does not, it is only useful for the IRepoistory, as it has a GetbyIdAsync method. Will update

**Pull request merged**